back/app/services/admin_publications_filtering_service.rb

```diff
@@ -20,23 +20,24 @@ def for_homepage_filter(scope)
     scope.where(publication_type: Project.name)
   end
 
-  # This filter removes AdminPublications that represent folders which contain *only* projects which should not be visible to the current user.
-  # Here we are concerned with 'visibility' in terms of the Project.visible_to attribute, which can have one of 3 values: public, groups or admins.
-  add_filter('remove_not_allowed_parents') do |visible_publications, options|
-    next visible_publications unless ['true', true, '1'].include? options[:remove_not_allowed_parents]
-
-    public_project_ids          = Project.publicly_visible.ids
-    public_project_publications = AdminPublication.includes(:parent).where(publication_id: public_project_ids).where.not(parent_id: nil)
-
-    visible_or_public_publication_ids = visible_publications.map(&:parent_id).concat(public_project_publications.map(&:parent_id)).compact.uniq
-
-    parents_with_visible_children = visible_publications.where(id: visible_or_public_publication_ids)
-    parents_without_any_children  = visible_publications.where(children_allowed: true, children_count: 0)
-    non_parents                   = visible_publications.where(children_allowed: false)
-
-    parents_with_visible_children.or(parents_without_any_children)
+  # This filter removes AdminPublications that represent folders,
+  # which contain *only* projects which should not be visible to the current user.
+  add_filter('remove_not_allowed_parents') do |scope, options|
+    next scope unless ['true', true, '1'].include? options[:remove_not_allowed_parents]
+
+    parents_with_visible_children = scope.where(id: scope
```

Comment:
I would maybe preserve some variables because this long line with multiple queries is a bit hard to follow.

Reply:
@brentguf Previously, you said

> admins/folder moderators also won't see folders with nothing but drafts in them anymore

But AFAIS draft projects can be in `scope`, which means their parents will be included in the result. 

Currently, we only filter out drafts from `publicly_visible` projects (see `Project.publicly_visible.not_draft.ids.uniq`). TBH I'm not sure why we handle public projects separately. I assume they should be included in `scope`.

Reply:
> But AFAIS draft projects can be in scope, which means their parents will be included in the result.

Yep, I changed it again to old behavior like I said to avoid introducing bugs.

I'll see if I can introduce a sensible variable name.

Reply:
TBH, looking at the old variable names, I find them more confusing than anything.

Reply:
And yes, I'm also pretty confused atm about this filter. I think it should only filter out folders with only drafts. Not sure if we even need publicly_visible.

I'll have a further look. But I don't want to get into rabbit holes and mostly keep things as they were.

Reply:
I suppose this piece of code could be improved in these areas:
1. Currently, it doesn't hide all draft projects. If a project is not public but visible to the current user, it'll be displayed. If the draft projects are not hidden, what's the purpose of this PR?
2. I don't understand why we handle public projects separately. Probably, it's a good chance to remove it as this time we've spent quite some time to understand how this code works.
3. After 2, the code could become clearer. But if not, I would split this long line into multiple ones using one/two variables.

If it's still not clear, let's have a call to discuss it. Or let me know if you want me to change it, I'll gladly help :slightly_smiling_face:

Reply:
(edited the first point)

Reply:
UPDATE: all three issues were addressed in the latest comments.